### PR TITLE
Knob: hide cursor on wheel event for .8s

### PR DESCRIPTION
**Pull request by ronso0**
While working on #11041 I got bothered by the mouse pointer covering the effect parameter value when using the mouse wheel to turn the knob.

This hides the pointer like for mouse move (drag) events. Atm it's useful only for effect parameter knobs, but for consistency it's applied to all knobs.
Timeout is .8 seconds, the same time the parameter value is shown

### CODE REVIEW

**Comment by daschuer** [comment]
Good idea.

This kind of confirms my concerns of changing the mouspointer to a hand.

**Comment by daschuer** [comment]
Is only the pointer hidden or is the mouse kind of catched like when turning the knob klick+move 
For my understanding we want the later?

**Comment by ronso0** [comment]
> This kind of confirms my concerns of changing the mouspointer to a hand.

Your comment confirms my impression you've been mixing things up ; )
This is about the **knob**, the hand cursor is for the parameter **label** only.

> Is only the pointer hidden or is the mouse kind of catched like when turning the knob klick+move
For my understanding we want the later?

Why would we want to lock the cursor position when scrolling? If you move the cursor while scrolling, that's okay, as soon as the cursor leaves the widget, the regular cursor is restored so it can always be located since.
(this reminds me to unset the cursor in LeaveEvent so it can't be hidden by hovering the knob during the timeout (when not scrolling))

**Comment by daschuer** [comment]
> Your comment confirms my impression you've been mixing things up ; )

Yes, you are right unfortunately. 
I think I need to test it tonight. 

I still do not understand how this solves the problem. When you move the mouse during scrolling and hit another widget it is hovered. 
That is distracting.

But before hovering another widget you leave the knob widget which bypassed the solution here.
Is this an issue?

**Comment by ronso0** [comment]
> (this reminds me to unset the cursor in LeaveEvent so it can't be hidden by hovering the knob during the timeout (when not scrolling))

done.

I think I'll also may implement lazy timer creation like for `ControlPushButtonBehavior` since the same reasoning applies to knobs, too.
https://github.com/mixxxdj/mixxx/blob/28e8b3fe2a09a17cd3841fc41f4e27b85e67df76/src/control/controlbehavior.h#L149-L156

**Comment by ronso0** [comment]
> When you move the mouse during scrolling and hit another widget it is hovered.
That is distracting.

I don't understand, please describe the distraction in that case.
Do you think users move the mouse wider than a knob widget while scrolling like mad?

Btw I think move events could of course be ignored while scrolling, though **not** while the cursor is still hidden some time _after_ scrolling, because then I'd expect the cursor move again (like instant release after a drag event).

File: src/widget/knobeventhandler.h [hunk 1]
```diff
@@ -1,22 +1,27 @@
...
+#include <QMouseEvent>
 #include <QPixmap>
+#include <QPoint>
+#include <QTimer>
+#include <QWheelEvent>
 
 #include "util/math.h"
 
+// duration the cursor is blanked after a mouse wheel event
+constexpr int wheelEventCursorTimeout = 2800;
```
Comment (daschuer) [inline src/widget/knobeventhandler.h:14]: 2.8 s feels long is this a debug leftover.

**Comment by daschuer** [comment]
I have just tested this PR and it works like described. Now I also understand the issue and how it is solved.

File: src/widget/knobeventhandler.h (hunk 1 above)
Comment (ronso0) [reply src/widget/knobeventhandler.h:14]: whoopsy... yes :roll_eyes:

**Comment by ronso0** [comment]
ping @daschuer

**Review by daschuer — CHANGES_REQUESTED** [review]
Sorry for the delay. A few more findings.

File: src/widget/knobeventhandler.h [hunk 2]
```diff
@@ -123,4 +153,5 @@ class KnobEventHandler {
     QPoint m_startPos;
     QPoint m_prevPos;
     QCursor m_blankCursor;
+    QTimer* m_wheelCursorTimer;
```
[the remark this answers is not in the document]
Comment (ronso0) [reply src/widget/knobeventhandler.h]: done.

File: src/widget/knobeventhandler.h [hunk 3]
```diff
@@ -113,6 +123,26 @@ class KnobEventHandler {
         pWidget->setControlParameter(newValue);
         pWidget->inputActivity();
         e->accept();
+        if (!m_wheelCursorTimer) {
+            m_wheelCursorTimer = new QTimer();
```
[the remark this answers is not in the document]
Comment (ronso0) [reply src/widget/knobeventhandler.h]: I use the pWidget as parent now.

File: src/widget/knobeventhandler.h [hunk 4]
```diff
@@ -1,22 +1,29 @@
...
   public:
     KnobEventHandler()
-            : m_bRightButtonPressed(false) {
-            QPixmap blankPixmap(32, 32);
-            blankPixmap.fill(QColor(0, 0, 0, 0));
-            m_blankCursor = QCursor(blankPixmap);
+            : m_bRightButtonPressed(false),
+              m_wheelCursorTimer(nullptr) {
+        QPixmap blankPixmap(32, 32);
+        blankPixmap.fill(QColor(0, 0, 0, 0));
```
[the remark this answers is not in the document]
Comment (ronso0) [reply src/widget/knobeventhandler.h]: works for me

**Review by daschuer — APPROVED** [review]
Thank you. Looks and works good.

**Pull request merged**